### DISCUSSION TITLE
SQLSTATE[HY000]: General error: 1364 Field 'name' doesn't have a default value (SQL: insert into `categories` (`updated_at`, `created_at`) values (2023-07-22 07:11:19, 2023-07-22 07:11:19))

### DISCUSSION THREAD

**Discussion by Danialdevs** [orchidsoftware/platform#2679] [Q&A]
SQLSTATE[HY000]: General error: 1364 Field 'name' doesn't have a default value (SQL: insert into `categories` (`updated_at`, `created_at`) values (2023-07-22 07:11:19, 2023-07-22 07:11:19))

public function layout(): iterable
    {
        return [
            Layout::table('categories', [
                TD::make('name', 'Имя')
            ]),
            Layout::modal('create', Layout::rows([
                Input::make('name')->required()->title('Имя'),
                Select::make('parent_id')->fromModel(category::class, 'name')->title('Под категория')
            ]))->title('Создать')->applyButton('Создать')

        ];
    }

    public function create(Request $request): void {
        $request->validate([
            'name' => ['required'],
            'parent_id' => ['exists:categories,id', 'required'] // Assuming 'categories' is the table name
        ]);

        Category::create([
            'name' => $request->name,
            'parent_id' => $request->parent_id,
        ]);
        Toast::info('Успешно');
    }

**Comment by surajtiwari06** [comment]
Please ensure that Your column must be default in Null in MySQL database

**Comment by BNohut** [comment]
I understood that your  'name' field of categories table assigned as 'nullable(false)' when migrated.
First - Check your fillable fields in Category model in App\Models\Categories.php
And be sure that you have defined your fields correctly.
` 
    protected $fillable = [
        'name',
        'parent_id',
    ];`
    Second-
    If everything is ok in model.php; then check the $request object that came in to 'create' function by using
    `dd($request);`
    If you can see your 'name' and 'parent_id' parameters in 'request>parameters' in $request. Then change your method of catching them in create function as
Third-
    ` Category::create([
        'name' => $request->get('name'),
        'parent_id' => $request->get('parent_id'),
    ]);`
    I hope theese advices will work for you.
